src/main/scala/com/chaordicsystems/cactus/Field.scala

```diff
@@ -0,0 +1,118 @@
+package com.chaordicsystems.cactus
+
+import com.chaordicsystems.cactus.Operator.Operator
+import com.sksamuel.elastic4s.ElasticDsl.{bool, matchQuery, must, nestedQuery, not, rangeQuery, should, termQuery}
+import com.sksamuel.elastic4s.QueryDefinition
+
+case class Field(value: String) extends UnaryOperation with MultiaryOperation {
+  private val isFieldNested: Boolean = value.contains(".")
+
+  private val getPath: String = value.split('.')(0)
+  private val getName: String = value.split('.')(1)
+
+  private def nestedOrElse(query: QueryDefinition): QueryDefinition =
+    if (isFieldNested) nestedQuery(getPath).query { query }
+    else query
+
+  private def comparativeQueryHandler(op: Operator, args: Any): QueryDefinition = {
+    op match {
+      case Operator.LT  => rangeQuery(value) to args includeUpper false
+      case Operator.GT  => rangeQuery(value) from args includeLower false
+      case Operator.LE  => rangeQuery(value) to args includeUpper true
+      case Operator.GE  => rangeQuery(value) from args includeUpper true
+      case Operator.NE  => matchQuery(value, args)
+      case Operator.EQ  => matchQuery(value, args)
+      case _            => throw InvalidOperatorException()
+    }
+  }
+
+  private def comparativeQuery(op: Operator, args: Any): QueryDefinition = {
+    args match {
+      case _: Boolean => termQuery(value, args)
+      case d: Double  => comparativeQueryHandler(op, d)
+      case b: BigInt  => comparativeQueryHandler(op, b.toInt)
```

Comment:
Why reducing number scope? No advantages, "you" already allocated memory for a `BigInt`. Everything should work with `BigInt`.